source/common/network/connection_impl.cc

```diff
@@ -730,19 +731,22 @@ void ConnectionImpl::flushWriteBuffer() {
 }
 
 ServerConnectionImpl::ServerConnectionImpl(Event::Dispatcher& dispatcher,
+                                           Server::ThreadLocalOverloadState& overload_state,
                                            ConnectionSocketPtr&& socket,
                                            TransportSocketPtr&& transport_socket,
                                            StreamInfo::StreamInfo& stream_info, bool connected)
     : ConnectionImpl(dispatcher, std::move(socket), std::move(transport_socket), stream_info,
-                     connected) {}
+                     connected),
+      overload_state_(overload_state) {}
 
 void ServerConnectionImpl::setTransportSocketConnectTimeout(std::chrono::milliseconds timeout) {
   if (!transport_connect_pending_) {
     return;
   }
   if (transport_socket_connect_timer_ == nullptr) {
     transport_socket_connect_timer_ =
-        dispatcher_.createTimer([this] { onTransportSocketConnectTimeout(); });
+        overload_state_.createScaledTimer(Server::OverloadTimerType::TransportSocketConnectTimeout,
+                                          [this] { onTransportSocketConnectTimeout(); });
```

Comment:
As mentioned elsewhere, I'd rather keep scaled timer creation going through the Overload Manager since it is ultimately responsible for determining the scale factor of those timers. We can revisit making the Dispatcher aware of the OM later.

Reply:
Instead of plumbing the OM widely, could the dispatcher hold a reference to it? I dislike that there are now two very different places to create timers: OM and dispatcher.

Reply:
The dispatcher is also owned by the thread.  I wonder if the solution would be to have the dispatcher own the thread overload state instead of having it be part of the thread local storage.

Reply:
> The dispatcher is also owned by the thread. I wonder if the solution would be to have the dispatcher own the thread overload state instead of having it be part of the thread local storage.

Yeah I think that makes sense.

Reply:
+1

Reply:
I started working on this. It's doable but introduces some weirdness; see #14401 for more.

Reply:
Made a fresh attempt at this in #14679 which feels reasonably clean.